### PR TITLE
 MULE-8207: Fix ConcurrentModificationException in FileMessageReceiver.

### PR DESCRIPTION
**Pull request by facundovs**
When no files are found in FileMessageReceiver polling, an empty list is returned.
However this empty list is mutable (an instance of ArrayList).
We should return an immutable list instead for avoiding Concurrent Exceptions in heavy load.

### CODE REVIEW

File: transports/file/src/test/java/org/mule/transport/file/FilePollEmptyResultTestCase.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+import org.mule.util.concurrent.Latch;
+
+import org.junit.Before;
+import org.junit.Test;
+
+
+public class FilePollEmptyResultTestCase extends FunctionalTestCase
+{
+
+    private static long TEST_TIMEOUT = 5000;
```
Comment (pablokraan) [inline transports/file/src/test/java/org/mule/transport/file/FilePollEmptyResultTestCase.java]: final

File: transports/file/src/test/java/org/mule/transport/file/FilePollEmptyResultTestCase.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+    private static Throwable concurrentException;
+
+    private Latch latch = new Latch();
```
Comment (pablokraan) [inline transports/file/src/test/java/org/mule/transport/file/FilePollEmptyResultTestCase.java]: final

File: transports/file/src/test/java/org/mule/transport/file/FilePollEmptyResultTestCase.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        openDirectory(getFileInsideWorkingDirectory("test").getAbsolutePath());
+        openDirectory(getFileInsideWorkingDirectory("test2").getAbsolutePath());
+        muleContext.setExceptionListener(new TestExceptionHandler());
+    }
+
+    @Test
+    public void testNoConcurrentExceptionWhenEmptyPollResult() throws InterruptedException
+    {
+        assertThat(latch.await(TEST_TIMEOUT, MILLISECONDS), is(false));
```
Comment (pablokraan) [inline transports/file/src/test/java/org/mule/transport/file/FilePollEmptyResultTestCase.java]: Is the test broken when the fix is not there?

Comment (facundovs) [reply transports/file/src/test/java/org/mule/transport/file/FilePollEmptyResultTestCase.java]: Yes. The test breaks consistently without the changes.

**Pull request merged**